src/Admin/Ajax.php

```diff
@@ -153,7 +153,17 @@ public function check_applepay_permissions() {
 		}
 
 		PayplugWoocommerceHelper::set_transient_data($account);
-		$applepay = isset($account['httpResponse']['payment_methods']['apple_pay']['enabled']) ? $account['httpResponse']['payment_methods']['apple_pay']['enabled']: false;
+		$applepay = false;
+
+		if ($account['httpResponse']['payment_methods']['apple_pay']['enabled']) {
+			foreach ($account['httpResponse']['payment_methods']['apple_pay']['allowed_domain_names'] as $domain_name) {
```

Comment:
Unfortunetly the "allowed_domain_names" check did'nt work; it always returns false in both localhost and in the specefic ngrok domain...
But this one is working:
```
$applepay = false;
	if ($account['httpResponse']['payment_methods']['apple_pay']['enabled']) {
		$domain_name = $_SERVER['SERVER_NAME'];
		if(in_array($domain_name, $account['httpResponse']['payment_methods']['apple_pay']['allowed_domain_names'])) {
			$applepay = true;
		}
	}
```

Reply:
As we discussed : 

- We should use wordpress defined methods 
- You should change your inputs in the general settings page, they must be the same as the shown URL
- We can never test with ngrok because it blocks ajax requests while our docker is not HTTPS, so you tested with ngrok?